src/avx512-16bit-qsort.hpp

```diff
@@ -548,25 +548,35 @@ avx512_qsort_fp16_helper(uint16_t *arr, arrsize_t arrsize)
     using T = uint16_t;
     using vtype = zmm_vector<float16>;
 
-#ifdef XSS_COMPILE_OPENMP
+#ifdef XSS_BUILD_WITH_STD_THREADS
     bool use_parallel = arrsize > 100000;
+#else
+    bool use_parallel = false;
+#endif
 
     if (use_parallel) {
-        // This thread limit was determined experimentally; it may be better for it to be the number of physical cores on the system
+#ifdef XSS_BUILD_WITH_STD_THREADS
+
+        // This thread limit was determined experimentally
         constexpr int thread_limit = 8;
-        int thread_count = std::min(thread_limit, omp_get_max_threads());
+        int thread_count = std::min(thread_limit,
+                                    (int)std::thread::hardware_concurrency());
```

Comment:
Why is 8 threads in a system with a lot of cores still better? Maybe you need to partition the data better to avoid locality problems.